examples/tutorials/asr_inference_with_cuda_ctc_decoder_tutorial.py

```diff
@@ -0,0 +1,303 @@
+"""
+ASR Inference with CUDA CTC Decoder
+====================================
+
+**Author**: `Yuekai Zhang <yuekaiz@nvidia.com>`__
+
+This tutorial shows how to perform speech recognition inference using a
+CUDA-based CTC beam search decoder. We demonstrate this on a pretrained `Zipformer <https://github.com/k2-fsa/icefall/tree/master/egs/librispeech/ASR/pruned_transducer_stateless7_ctc>`__ model from next-gen-kaldi project.
+
+"""
+
+######################################################################
+# Overview
+# --------
+#
+# Beam search decoding works by iteratively expanding text hypotheses (beams)
+# with next possible characters, and maintaining only the hypotheses with the
+# highest scores at each time step. 
+#
+# The underlying implementation uses cuda to acclerate the whole decoding process. A mathematical formula for the decoder can be
+# found in the `paper <https://arxiv.org/pdf/1408.2873.pdf>`__, and
+# a more detailed algorithm can be found in this `blog
+# <https://distill.pub/2017/ctc/>`__.
+#
+# Running ASR inference using a CUDA CTC Beam Search decoder
+# requires the following components
+#
+# -  Acoustic Model: model predicting modeling units (BPE in this tutorial) from acoustic features
+# -  BPE Model: the byte-pair encoding (BPE) tokenizer file
+#
+
+
+######################################################################
+# Acoustic Model and Set Up
+# -------------------------
+#
+# First we import the necessary utilities and fetch the data that we are
+# working with
+#
+
+import torch
+import torchaudio
+
+print(torch.__version__)
+print(torchaudio.__version__)
+
+######################################################################
+#
+
+import time
+from typing import List
+
+import IPython
+import matplotlib.pyplot as plt
+from torchaudio.models.decoder import cuda_ctc_decoder
+from torchaudio.utils import download_asset
+import os
+
+######################################################################
+#
+# We use the pretrained `Zipformer <https://huggingface.co/Zengwei/icefall-asr-librispeech-pruned-transducer-stateless7-ctc-2022-12-01>`__
+# model that is trained on the `LibriSpeech
+# dataset <http://www.openslr.org/12>`__. The model is jointly trained with CTC and Transducer loss functions.
+# In this tutorial, we only use CTC head of the model.
+
+model_link="https://huggingface.co/Zengwei/icefall-asr-librispeech-pruned-transducer-stateless7-ctc-2022-12-01/resolve/main/exp/cpu_jit.pt"
+os.system(f"wget -nc  {model_link}")
```

Comment:
We cannot assume all the machines have wget, so could you use `torch.hub.download_url_to_file(url, path)` ? 

Then for the download location, can you use `Path(torch.hub.get_dir()) / "torchaudio" / "<some_discriptive_name>")`